### PR TITLE
root: integrated C programming language for its library construction

### PR DESCRIPTION
**Pull request by hollowaykeanho**
Since the hardest library construction is C programming language, we should deal with it first. Hence, let's do this.

This patch integrates C programming language for its library construction in root repository.

### CODE REVIEW

File: .github/workflows/git-push.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
         CONTAINER_PASSWORD: ${{ secrets.CONTAINER_PASSWORD }}
         TWINE_USERNAME: ${{ secrets.TWINE_USERNAME }}
         TWINE_PASSWORD: ${{ secrets.TWINE_PASSWORD }}
-        PROJECT_SIMULATE_RELEASE_REPO: true
+        PROJECT_SIMULATE_RUN: true
         PROJECT_ROBOT_RUN: true
```
Comment (corygalyna) [inline .github/workflows/git-push.yml:32]: Mark this as non-backward compatible.

File: automataCI/services/io/os.sh [hunk 2]
```diff
@@ -133,7 +133,7 @@ OS_Is_Command_Available() {
 
 OS_Is_Run_Simulated() {
         # execute
-        if [ ! -z "$PROJECT_SIMULATE_RELEASE_REPO" ]; then
+        if [ ! -z "$PROJECT_SIMULATE_RUN" ]; then
                 printf -- "0"
```
Comment (corygalyna) [inline automataCI/services/io/os.sh:137]: Checked. Safe to implement flag changes.

**Review by corygalyna — APPROVED** [review]
Finally done. Please avoid such a big patch in the future. It's really hard to review them especially after work and near mid-night. :(

**Comment by corygalyna** [comment]
By the way, where is MSI packager?

**Comment by corygalyna** [comment]
And also, in some codes, I assume this patch will implement flat package repository feature?

File: .github/workflows/git-push.yml (hunk 1 above)
Comment (hollowaykeanho) [reply .github/workflows/git-push.yml:32]: Good evening over there. Noted.

**Comment by hollowaykeanho** [comment]
> By the way, where is MSI packager?

Will implement in the next patch. It has this issue https://github.com/wixtoolset/issues/issues/7896 to co-implement as well.

**Comment by hollowaykeanho** [comment]
> And also, in some codes, I assume this patch will implement flat package repository feature?

yes

**Comment by corygalyna** [comment]
OK. In that case, all clear. You may also want to push to upstream (AutomataCI repo) before I go to bed.

**Pull request merged**